scalding-core/src/test/scala/com/twitter/scalding/TypedSketchJoinJobForEmptyKeysTest.scala

```diff
@@ -0,0 +1,35 @@
+package com.twitter.scalding
+
+import org.scalatest.{ WordSpec, Matchers }
+
+class TypedSketchJoinJobForEmptyKeys(args: Args) extends Job(args) {
+  // Deal with when a key appears in left but not right
+  val leftTypedPipe = TypedPipe.from(List((1, 1111)))
+  val rightTypedPipe = TypedPipe.from(List((3, 3333), (4, 4444)))
+
+  implicit def serialize(k: Int) = k.toString.getBytes
+  leftTypedPipe
+    .sketch(1)
+    .leftJoin(rightTypedPipe)
+    .map{
+      case (a, (b, c)) =>
+        (a, b, c.getOrElse(-1))
+    }
+    .write(TypedTsv("output"))
+}
+
+class TypedSketchJoinJobForEmptyKeysTest extends WordSpec with Matchers {
+  import Dsl._
+  "A TypedSketchJoinJobForEmptyKeysTest" should {
+    "Sketch leftJoin should be the same an Join which is empty" in {
```

Comment:
I didn't edit it properly. There was originally a normal join and a sketch join in the job, but I removed the normal join since it was just easier to assert the output was a particular thing. This particular input and output caused an exception before, so the fact that it is green now is a guard against a particular regression, but the message is not good. I'll update it.